lib/InjectionException.php

```diff
@@ -14,24 +14,59 @@ public function __construct(array $inProgressMakes, $message = "", $code = 0, \E
         parent::__construct($message, $code, $previous);
     }
 
+    public static function fromInvalidDefineParamsNotArray($definition, array $inProgressMakes): self
+    {
+        $message = sprintf(
+            Injector::M_INVALID_DEFINE_ARGUMENT_NOT_ARRAY,
+            gettype($definition)
+        );
+
+        return new self(
+            $inProgressMakes,
+            $message,
+            Injector::E_INVALID_DEFINE_ARGUMENT_NOT_ARRAY
+        );
+    }
+
+    public static function fromInvalidDefineParamsBadKeys($definition, array $inProgressMakes)
```

Comment:
```suggestion
    public static function fromInvalidDefineParameterKeys($definition, array $inProgressMakes)
```

We don't need to repeat `Bad` here, because we already have `Invalid` in front.